R/dataPrepRast.R

```diff
@@ -0,0 +1,131 @@
+#' Crop, reproject, and mask a raster map
+#'
+#' This script reprojects and crop a raster using a raster or polygon object.
+#'
+#' @param x   A \code{RasterLayer} object or the name of one.
+#' @param cropWith   Extent object, or any object from which an Extent object can
+#'                   be extracted (see Details of \code{\link[raster]{crop}}).
+#' @params maskWith   Raster* or Spatial object passed to \code{\link[raster]{mask}}.
+#' @params reprojectTo   Raster* or Spatial object passed to \code{\link[raster]{crs}}.
+#' @params outPath   Path where fileOut is saved.
+#' @params fileOut   File name to write to (\code{\link[raster]{writeRaster}}).
+#' @params formatOut File format to write out (\code{\link[raster]{writeRaster}}).
+#'                   If not specified, GTiff will be used.
+#' @params timings   Logical indicating whether Sys.time messages should be
+#'                   suppressed. Default is \code{FALSE}.
+#' @param ...   Additional arguments to the aforementioned \code{raster} functions.
+#'
+#' @return Invisibly returns the cropped, projected, and masked raster.
```

Comment:
why invisibly? this should not just return a `RsaterLayer` but should work on stacks etc. too.